app/services/metadata_templates/create_service.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module MetadataTemplates
+  # Service used to Create Metadata Templates
+  class CreateService < BaseService
+    MetadataTemplateCreateError = Class.new(StandardError)
+    attr_accessor :namespace
+
+    def initialize(user, namespace, params = {})
+      super(user, params)
+      @metadata_template = nil
+      @namespace = namespace
+      @params = params
+    end
+
+    def execute
+      authorize! namespace, to: :create_metadata_template?
+
+      validate_params
+
+      build_template
+      save_template
+      @metadata_template
+    rescue MetadataTemplates::CreateService::MetadataTemplateCreateError => e
+      @namespace.errors.add(:base, e.message)
+    end
+
+    private
+
+    def validate_params
+      if @params[:name].blank?
+        raise MetadataTemplateCreateError,
+              I18n.t('services.metadata_templates.create.required.name')
+      end
+
+      return unless @params[:fields].blank? || !@params[:fields].is_a?(Array)
+
+      raise MetadataTemplateCreateError,
+            I18n.t('services.metadata_templates.create.required.fields')
+    end
+
+    def build_template
+      @metadata_template = MetadataTemplate.new(@params.merge(
+                                                  created_by: current_user,
+                                                  namespace: @namespace
+                                                ))
+    end
+
+    def save_template
+      return unless @metadata_template.save
+
+      create_activities
+    end
+
+    def can_create_template?
```

Comment:
I don't think this method is used so it can be removed

Reply:
Removed in a487903b64c50014af27bed9e5ae06a698976534

Reply:
Removed in a487903b64c50014af27bed9e5ae06a698976534